freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaStartService.java

```diff
@@ -0,0 +1,40 @@
+package com.sequenceiq.freeipa.service.stack;
+
+import static com.sequenceiq.freeipa.flow.stack.start.StackStartEvent.STACK_START_EVENT;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.freeipa.controller.exception.NotFoundException;
+import com.sequenceiq.freeipa.entity.Stack;
+import com.sequenceiq.freeipa.flow.stack.StackEvent;
+import com.sequenceiq.freeipa.service.freeipa.flow.FreeIpaFlowManager;
+import com.sequenceiq.freeipa.service.stack.instance.InstanceMetaDataService;
+
+@Service
+public class FreeIpaStartService {
+
+    @Inject
+    private StackService stackService;
+
+    @Inject
+    private FreeIpaFlowManager flowManager;
+
+    @Inject
+    private StackUpdater stackUpdater;
+
+    @Inject
+    private InstanceMetaDataService instanceMetaDataService;
+
+    public void start(String environmentCrn, String accountId) {
+        List<Stack> stacks = stackService.findAllByEnvironmentCrnAndAccountId(environmentCrn, accountId);
+        if (stacks.isEmpty()) {
```

Comment:
Repos might return nulls

Reply:
Hm, I don't understand this. Which repo?

Reply:
Sorry, I meant JpaRepositories. I might be wrong, but either way, using CollectionUtils.isEmpty() won't hurt.